src/middle/Stmt.ml

```diff
@@ -25,7 +25,8 @@ module Fixed = struct
           { decl_adtype: UnsizedType.autodifftype
           ; decl_id: string
           ; decl_type: 'a Type.t
-          ; initialize: bool }
+          ; initialize: bool
+          ; assignment: 'a option }
```

Comment:
`initialize` is irrelevant if `assignment` is set, right? I'd rather change this to
```suggestion
          ; initialize: 'a decl_init }
```
with
```ocaml
type 'a decl_init = Uninit | Default | Assign of 'a
```
where the first one means `initialize=false`, the second means `initialize=true` and the last one means `assignment=Some`.

Reply:
I agree with the suggestion, but I wonder if it couldn't go further? The optimization detects an assignment in order to set `initialize`, so couldn't it just move the assignment inside the decl ?

Reply:
ooh thats a pretty way to do this! I'll update

Reply:
> I agree with the suggestion, but I wonder if it couldn't go further? The optimization detects an assignment in order to set initialize, so couldn't it just move the assignment inside the decl ?

I agree with this but think having all the decls inline will be enough changes to have a separate PR